### PR TITLE
Add marble diagrams to a few Single.doOnX methods.

### PR DESCRIPTION
**Pull request by VeskoI**
This should tick a few tasks from this issue: https://github.com/ReactiveX/RxJava/issues/5788.

The urls are a bit ugly, as they're from the uploads in the issue. I'm happy to update the urls with different ones if there's a better place to upload the diagrams.

### CODE REVIEW

**Comment by akarnokd** [comment]
Thanks for the PR. There are a couple of things to be changed on the diagrams. Once done, I'll upload them into this repo and provide you with the proper URLs so you can update this PR.

**Comment by VeskoI** [comment]
Sure, will get these fixed tonight :)

File: src/main/java/io/reactivex/Single.java [hunk 1]
```diff
@@ -2173,6 +2173,9 @@
      * is executed once per subscription.
      * <p>Note that the {@code onFinally} action is shared between subscriptions and as such
      * should be thread-safe.
+     * <p>
+     * <img width="1040" height="472" src="https://user-images.githubusercontent.com/4089934/39448132-dca93836-4cbb-11e8-9b13-ce078a8ba206.png" alt="">
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Single.java]: Images should be 640 width and the height should be reduced to keep the aspect ratio. In this case, 2079x945 -> 640x290

**Comment by VeskoI** [comment]
Hope these are better:

single_dofinally
single_doondispose
single_doonsubscribe

**Comment by akarnokd** [comment]
Great, here are the links:

https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doFinally.png
https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnSuccess.2.png
https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnError.2.png
https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnDispose.png
https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/Single.doOnSubscribe.png

Some images existed but we don't change those to avoid conflicts with the ReactiveX site linking to them.

**Review by akarnokd — APPROVED** [review]

**Review by vanniktech — APPROVED** [review]

**Pull request merged**